src/pybind/mgr/dashboard/frontend/src/app/shared/components/download-button/download-button.component.html

```diff
@@ -8,16 +9,30 @@
   </button>
   <div ngbDropdownMenu>
     <button ngbDropdownItem
-            (click)="download('json')"
-            *ngIf="objectItem">
+            (click)="download('json')">
       <i [ngClass]="[icons.json]"></i>
       <span>JSON</span>
     </button>
     <button ngbDropdownItem
-            (click)="download()"
-            *ngIf="textItem">
+            (click)="download()">
       <i [ngClass]="[icons.text]"></i>
       <span>Text</span>
     </button>
   </div>
 </div>
+<div placement="bottom-right">
+  <button type="button"
+          [title]="title"
+          class="btn btn-light dropdown-toggle-split"
+          *ngIf="textItem && !objectItem"
+          (click)="download()">
+    <i [ngClass]="[icons.download]"></i>
+  </button>
+  <button type="button"
+          [title]="title"
+          class="btn btn-light dropdown-toggle-split"
+          *ngIf="objectItem && !textItem"
+          (click)="download('json')">
+    <i [ngClass]="[icons.download]"></i>
+  </button>
+</div>
```

Comment:
There is a small issue here, the `div placement="bottom-right">` will always be rendered since it is not encapsulated within any ngIf

A proposal of mine, to avoid some of the current repetition as well:

```suggestion
<ng-template #simpleButton>
  <div placement="bottom-right"
      *ngIf="textItem || objectItem">
    <button type="button"
            [title]="title"
            class="btn btn-light dropdown-toggle-split"
            (click)="download(textItem ? '' : 'json')">
      <i [ngClass]="[icons.download]"></i>
    </button>
  </div>
</ng-template>
</div>
```